src/smolagents/models.py

```diff
@@ -377,7 +391,26 @@ def __call__(
         Returns:
             `ChatMessage`: A chat message object containing the model's response.
         """
-        pass  # To be implemented in child classes!
+        raise NotImplementedError("This method must be implemented in child classes")
```

Comment:
What about defining Model as an abstract class and decorating this method as abstractmethod?